### DISCUSSION TITLE
use local instance of drawio instead of diagrams.net

### DISCUSSION THREAD

**Discussion by Keskejefaislamoi** [requarks/wiki#6196] [Error / Bug Report]
hi,

I can't find where I have to specify the use of a local instance of diagrams.net (drawio)

My instance off wiki.js is offline, i cant insert diagrams without local instance of draw.io

i cannot found anything in the config file : /etc/wiki/config.yml

thanks

**Comment by Philipinho** [comment]
The line you are looking for is here.
https://github.com/requarks/wiki/blob/3a0ec8d7fcd9c7b74591394ffa1908088e635770/client/components/editor/editor-modal-drawio.vue#L5

Replace `embed.diagrams.net` with your own local drawio instance url.
Note that updates will override any changes.
If you use docker, you can bind mount the file so as not to worry about updates overriding it.

**Reply by Keskejefaislamoi** [reply]
thanks for your response,
i dont have : client/components/editor/editor-modal-drawio.vue
when i make an :
grep -Ril 'embed.diagrams.net' /etc/wiki

i got :  /etc/wiki/assets/js/editor.js

i have try to edit the file  /etc/wiki/assets/js/editor.js 
embed.diagrams.net ---> myinstance.local

systemctl restart wiki

the instance still tries to load from : diagrams.net, what im doing wrong ?

best

**Reply by tfboy** [reply]
> The line you are looking for is here.
> 
> https://github.com/requarks/wiki/blob/3a0ec8d7fcd9c7b74591394ffa1908088e635770/client/components/editor/editor-modal-drawio.vue#L5
> 
> Replace `embed.diagrams.net` with your own local drawio instance url. Note that updates will override any changes. If you use docker, you can bind mount the file so as not to worry about updates overriding it.

I'm confused (and not an experienced github user, so please pardon me!) - the download tarball doesn't contain all the files referenced in github. It includes the ./server directory, but not the ./client one which has this file that would need editing.

I'm super keen to see if this can work as I have been attempting to get draw.io integration with an "offline" instance using Kroki but without much success.

**Reply by Philipinho** [reply]
The tar is built already. If you make any changes to the source, you have to build it yourself.

**Reply by tfboy** [reply]
Ah that's what I wondered.
So there's no easy way to fix / edit the standard install to make it work? You have to start from scratch and compile / build incorporating the changes?
I know there was another thread with discussion about using self-hosted drawio / diagrams.net instead of the public one but it wasn't possible.
If there's no easy change and a complete build is required, then that might make sense, even if it seems a shame.

**Reply by tfboy** [reply]
> thanks for your response, i dont have : client/components/editor/editor-modal-drawio.vue when i make an : grep -Ril 'embed.diagrams.net' /etc/wiki
> 
> i got : /etc/wiki/assets/js/editor.js
> 
> i have try to edit the file /etc/wiki/assets/js/editor.js embed.diagrams.net ---> myinstance.local
> 
> systemctl restart wiki
> 
> the instance still tries to load from : diagrams.net, what im doing wrong ?
> 
> best

Actually, having had another look at this, editing the file ./assets/js/editor.js and replacing "embed.diagrams.net" with your own URL does work ! But you do have to clear your browser cache or try in a new private browsing window. This also seems to work for me! As this is an https connection, you will need a matching valid certificate for it to work I think.

**Comment by tfboy** [comment]
An alternative solution, which I admit is a bit of a bodge, but I have tested it and it works: you create a local hosts file entry redirecting the FQDN "embed.diagrams.net" to a local server running the dockerised version of it.

Using https://github.com/jgraph/docker-drawio I have downloaded it to my server running WikiJS. I have the port mapping to port 8080 so it doesn't conflict with WikiJS listening on port 3000 nor Kroki on ports 8000-8005.

I'm using Nginx as a reverse proxy to terminate my connections so all traffic is redirected to port 3000 for WikiJS. I've now changed the config to be a server block (or virtual host if you're using Apache) and added a new server config for the drawio container. So http://wiki.local (as an example) points to my WikiJS server on port 80 which is locally redirected to port 3000 where WikiJS is running. Also, http://embed.diagrams.net also points to my WikiJS server, and with the new server block in Nginx, locally redirects to port 8080 where drawio container is running.

I've also added some SSL certificates (I have a local PKI so have issued them both for my wiki.local and embed.diagrams.net) and it works absolutely fine. This allows me to run WikiJS in SSL mode, and more importantly, allows the "burnt in" https://embed.diagrams.net URL to work correctly in SSL too.

This is a bit of a bodge and using my local PKI, I'm impersonating the real diagrams.net webserver and substituting it for my local docker container, but it works. At least until we can find an easy way of modifying the URL used by WikiJS when inserting diagrams.

**Reply by bendathierrycom** [reply]
Maybe setting the env variable within your docker command line to update all js related scripts : 
`-e PUBLIC_DNS=draw.yourdomain.com`

Did you try it ?

**Comment by chary1112004** [comment]
Hi, 

Could you integrate success from application with local instance of drawio already by replacing **embed.diagrams.net** with your own domain?
src='https://embed.diagrams.net/?embed=1&proto=json&spin=1&saveAndExit=1&noSaveBtn=1&noExitBtn=0'

We are integrating also from our application to self-host drawio however we saw it is keeping loading
![image](https://github.com/user-attachments/assets/f7249871-e0f7-4c8f-b11d-16150986bee7)
